### PR TITLE
creatures: refactor animation switches

### LINKED ISSUE #807: Refactor: use animation switching idiom on all objects
Replicate the method of switching animations introduced by #802 to all objects i.e.

```
item->anim_number = ANIM_NUM;
item->frame_number = FRAME_NUM;
```
becomes

`Item_SwitchToAnim(item, ANIM_NUM, FRAME_NUM);`

### PR DESCRIPTION
**Pull request by lahm86**
Resolves #807.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This brings animation switching for creatures into line with the same for Lara, as was introduced in #802.

### CODE REVIEW

File: src/game/objects/creatures/statue.c [hunk 1]
```diff
@@ -46,8 +48,8 @@ void Statue_Initialise(int16_t item_num)
 
     Item_Initialise(centaur_item_num);
 
-    centaur->anim_number = g_Objects[O_CENTAUR].anim_index + 7;
-    centaur->frame_number = g_Anims[centaur->anim_number].frame_base + 36;
+    Item_SwitchToAnim(centaur, CENTAUR_REARING_ANIM, -1);
+    centaur->frame_number += CENTAUR_REARING_FRAME;
```
Comment (rr-) [inline src/game/objects/creatures/statue.c]: Shouldn't this be
```
    Item_SwitchToAnim(centaur, CENTAUR_REARING_ANIM, CENTAUR_REARING_FRAME);
```
?

**Review by rr- — APPROVED** [review]
LGTM apart from one cosmetic comment

File: src/game/objects/creatures/statue.c (hunk 1 above)
Comment (lahm86) [reply src/game/objects/creatures/statue.c]: Yes, a silly mistake. Now resolved.

**Review by rr- — APPROVED** [review]

**Comment by rr-** [comment]
Please remember to squash the commit when merging

**Comment by walkawayy** [comment]
I thought maybe this was only for creatures, but since you included the rolling ball trap I'll ask. Should we update the call in [thors_hammer](https://github.com/LostArtefacts/Tomb1Main/blob/c1bcf9823206b1caeeed78349265bff8ac1ba213/src/game/objects/traps/thors_hammer.c#L154)?

There also seems to be two remaining calls in the game using the old method which are `O_LARA_EXTRA` related calls in [midas_touch](https://github.com/LostArtefacts/Tomb1Main/blob/c1bcf9823206b1caeeed78349265bff8ac1ba213/src/game/objects/traps/midas_touch.c#L45) and [trex](https://github.com/LostArtefacts/Tomb1Main/blob/c1bcf9823206b1caeeed78349265bff8ac1ba213/src/game/objects/creatures/trex.c#L196).

**Comment by lahm86** [comment]
> I thought maybe this was only for creatures, but since you included the rolling ball trap I'll ask. Should we update the call in [thors_hammer](https://github.com/LostArtefacts/Tomb1Main/blob/c1bcf9823206b1caeeed78349265bff8ac1ba213/src/game/objects/traps/thors_hammer.c#L154)?
> 
> There also seems to be two remaining calls in the game using the old method which are `O_LARA_EXTRA` related calls in [midas_touch](https://github.com/LostArtefacts/Tomb1Main/blob/c1bcf9823206b1caeeed78349265bff8ac1ba213/src/game/objects/traps/midas_touch.c#L45) and [trex](https://github.com/LostArtefacts/Tomb1Main/blob/c1bcf9823206b1caeeed78349265bff8ac1ba213/src/game/objects/creatures/trex.c#L196).

Those are a bit different as the object number doesn't match the item e.g. we're setting Lara's animation to the one specified in the `O_LARA_EXTRA` model, so we couldn't use the shared function as it pulls it from the item's object ID (so it'd be Lara). I think these special cases will have to remain, unless we make a shared and specific function for those?

**Comment by walkawayy** [comment]
Ah got it.

**Review by walkawayy — APPROVED** [review]

**Pull request merged**